src/packageurl/contrib/url2purl.py

```diff
@@ -263,3 +263,16 @@ def build_nuget_purl(uri):
 @purl_router.route(sourceforge_pattern)
 def build_sourceforge_purl(uri):
     return purl_from_pattern('sourceforge', sourceforge_pattern, uri)
+
+
+crates_pattern= (
+    r"^https?://crates.io/api/v1/crates/"
+    r"(?P<name>.+)/(?P<version>.+)"
+    r"(\/download)$"
+)
```

Comment:
I am not sure we want to rely on regex for constructing/deconstructing purls

Reply:
Hi, regex has been used above for Ruby, nugget , pypi and sourceforge so I think it is good to go

Reply:
fair enough.